container/client.go

```diff
@@ -118,17 +119,49 @@ func (client dockerClient) StartContainer(c Container) error {
 	bg := context.Background()
 	config := c.runtimeConfig()
 	hostConfig := c.hostConfig()
+	networkConfig := &network.NetworkingConfig{EndpointsConfig: c.containerInfo.NetworkSettings.Networks}
+	// simpleNetworkConfig is a networkConfig with only 1 network.
+	// see: https://github.com/docker/docker/issues/29265
+	simpleNetworkConfig := func() *network.NetworkingConfig {
+		oneEndpoint := make(map[string]*network.EndpointSettings)
+		for k, v := range networkConfig.EndpointsConfig {
+			oneEndpoint[k] = v
+			// we only need 1
+			break
+		}
+		return &network.NetworkingConfig{EndpointsConfig: oneEndpoint}
+	}()
+
 	name := c.Name()
 
 	log.Infof("Starting %s", name)
-	creation, err := client.api.ContainerCreate(bg, config, hostConfig, nil, name)
+	creation, err := client.api.ContainerCreate(bg, config, hostConfig, simpleNetworkConfig, name)
 	if err != nil {
 		return err
 	}
 
 	log.Debugf("Starting container %s (%s)", name, creation.ID)
 
-	return client.api.ContainerStart(bg, creation.ID, types.ContainerStartOptions{})
+	err = client.api.ContainerStart(bg, creation.ID, types.ContainerStartOptions{})
```

Comment:
@dolanor I've just seen that you reattach the networks after the container has already been started. I think it would be better if we first reattach the networks and afterwards start the container again. Or did you have some problems and made it in this order?